clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -460,6 +460,28 @@ static void FixupInvocation(CompilerInvocation &Invocation,
     LangOpts.NewAlignOverride = 0;
   }
 
+  // Check whether the SYCL arguments are valid. -fsycl is only accepted in -cc1
+  // if it is passed in conjunction with -fsycl-is-device or -fsycl-is-host. If
+  // -fsycl is not passed but one of those two options is, automatically set it.
```

Comment:
IMO, I'd prefer fsycl just NOT work in cc1 mode.

Reply:
Do you know of any options in `Options.td` that do this? I tried to go down that route but I could find no way to get a driver-only option to work (mostly because it would never set `LangOpt.SYCL`). Related, does that mean you want `LangOpts.SYCL` to go away (because it's not a language option at that point, just a driver option)?

Reply:
I looked through a couple, and see `OPT_driver_mode` which I think is only meaningful to the driver?  Same as `-c` or `-###`:

```
bash-4.2$ ./bin/clang --driver-mode=gcc c6.cpp -std=c++17 -c
bash-4.2$ ./bin/clang -cc1 --driver-mode=gcc c6.cpp -std=c++17 -c
error: unknown argument: '--driver-mode=gcc'
error: unknown argument: '-c'
```


I think `LangOpt.SYCL` is still useful as Host OR Device, but I'm not sure how often we care about 'any' vs a specific host/device.

Reply:
As for the setting of `LangOpts.SYCL`, my thought is to make it just SYCLIsHost || SYCLIsDevice

Reply:
I think I actually prefer getting rid of `LangOpts.SYCL` entirely. It's used ~5 times in the frontend and only twice is it used to mean host or device mode. I think it's better to require people to think about host vs device instead of accidentally testing for SYCL in general.

Reply:
SGTM!

Reply:
@erichkeane -- You asked for it, now you can have 219 changed files in a patch you'll be reviewing twice. ;-)

I still need some assistance with resolving the CodeGenSYCL/loop_unroll.cpp failure.